prdoc/pr_7723.prdoc

```diff
@@ -0,0 +1,22 @@
+title: '[pallet-bounties] Make BountyUpdatePeriod optional'
+doc:
+- audience: Runtime Dev
+  description: |
+    Refactored `BountyUpdatePeriod` to be an optional configuration parameter.
+    If set to `None`, bounties remain active indefinitely, removing the need for
+    `extend_bounty_expiry` and preventing automatic curator slashing for inactivity.
```

Comment:
This could have one sentence for how to migrate to this new change. I assume that many people want to keep the old behaviour, so just please one mention for "how to keep the old behaviour" and "how to use this feature".